Seminars/Sem.01/Pract.01/task03.cpp

```diff
@@ -0,0 +1,68 @@
+#include <iostream>
+
+using namespace std;
+
+int **createMatrix(const int rows, const int cols) {
+    int **matrix = new int *[rows];
+    for (int i = 0; i < rows; i++) {
+        matrix[i] = new int[cols];
+    }
+    return matrix;
+}
+
+void inputMatrix(int **matrix, const int rows, const int cols) {
+    for (int i = 0; i < rows; i++) {
+        for (int j = 0; j < cols; j++) {
+            cout << "Enter the element at position (" << i << ", " << j << "): ";
+            cin >> matrix[i][j];
+        }
+    }
+}
+
+int **transposeMatrix(int **matrix, const int rows, const int cols) {
+    int **transposedMatrix = createMatrix(cols, rows);
+    for (int i = 0; i < rows; i++) {
+        for (int j = 0; j < cols; j++) {
+            transposedMatrix[j][i] = matrix[i][j];
+        }
+    }
+    return transposedMatrix;
+}
+
+void printMatrix(int **matrix, const int rows, const int cols) {
```

Comment:
Should we use const?

Reply:
No, the function call `printMatrix(matrix, rows, cols)` doesn't recognize `void printMatrix(const int **matrix, const int rows, const int cols)` because the passed parameter `matrix` isn't `const int**`.